docs/src/main/asciidoc/security-openid-connect-multitenancy-concept.adoc

```diff
@@ -0,0 +1,317 @@
+Multi-tenant applications can enable development teams to quickly provide and maintain functionality to many organisations without compromising security concerns. However, understanding and translating concepts and best practises from the OIDC world can be a daunting task, so this overview will illustrate the workings of a typical multi-tenant application to highlight the OIDC and security-related parts.
+
+.Two clients calling a multi-tenant application
+[caption="Figure 1: "]
+image::oidc-multitenancy-concept-1.svg[Multitenant Application]
+
+The image above shows 2 users from different organisations call a calendar service to find their respective organisations' festivities. For the sake of simplicity, we assume that the application serves RESTful data to a browser-based Single-Page Application (SPA).
+
+. Each user has logged in using their organisation's own single-sign ("SSO"), which is frequently called OIDC Authentication Provider (called `"OpenID Provider"`, `"Identity Provider"` or `"IdP"` in OIDC literature). Upon login, the SSO has produced an `IdToken` JWT with user data and delivered that to the user's web browser.
+. The two user will call the calendar service with _different resource paths_ within the service URL to reach the calendar service. The application uses the `tenant identifier` part of the URI to identify the caller's organisation. Should an authenticated user from organisation 1 accidentally use a resource path allocated to organisation 2, the call will yield an unauthenticated HTTP response.
+. The `IdToken` data retreived from the own organisation's SSO is normally sent to the multi-tenant application with each call as an HTTP header, called `"authorization:"`. In this respect, an IdToken works as a session cookie which contains user information.
+
+With every call, the multi-tenant application can translate the `tenant identifier` "org1" to imply that the user is logged in to the SSO of organisation 1, and "org2" to the SSO of organisation 2. The multi-tenant application can also verify that the user information in the `IdToken` is not changed, and extract the data within it.
+
+
+== The 20-minute overview of a multi-tenant application
+
+Quarkus multi-tenant applications need to perform several tasks with each call from a client to provide a simple and consistent development experience. Understanding the sequence of those steps helps designing multi-tenant applications correctly; the conceptual models described in the OIDC and OAuth2 specifications is huge and illustrating the steps required by development teams helps to clarify multi-tenant application development.
+
+In this example, our multi-tenant application is designed to serve RESTful resources to already logged-in users. This mode of operation is called `Bearer authentication` in OIDC and indicates that an HTTP header called `"authorization:"` with the value `"Bearer *encoded JWT token*"` must be present in each request from the client to the application.
```

Comment:
A good number of providers (Auth0, Google, Github, etc) would only have opaque/binary access tokens. So I'd clarify it